sphinxval/utils/duplicates.py

```diff
@@ -0,0 +1,417 @@
+
+""" utils/resume.py contains subroutines to aid in resuming
+    the validation process from a starting dataframe.
+    
+"""
+
+#Create logger
+logger = logging.getLogger(__name__)
+
+
+def initialize_forecast_dict():
+    """ Set up a dictionary for a pandas df to hold each possible quantity
+        stored in Forecast objects.
+        
+    """
+    #Convert to Pandas dataframe
+    #Include triggers with as much flattened info
+    #If need multiple dimension, then could be used as tooltip info
+    #Last CME, N CMEs, Last speed, last location, Timestamps array of all CMEs used
+    
+
+    dict = {"Model": [],
+            "Energy Channel Key": [],
+            "Prediction Index": [],
+            "All Thresholds in Prediction": [],
+            "Forecast Source": [],
+            "Forecast Path": [],
+            "Forecast Issue Time":[],
+            "Prediction Window Start": [],
+            "Prediction Window End": [],
+            
+            "Number of CMEs": [],
+            "CME Start Time": [], #Timestamp of 1st
+                #coronagraph image CME is visible in
+            "CME Liftoff Time": [], #Timestamp of coronagraph
+                #image with 1st indication of CME liftoff (used by
+                #CACTUS)
+            "CME Latitude": [],
+            "CME Longitude": [],
+            "CME Speed": [],
+            "CME Half Width": [],
+            "CME PA": [],
+            "CME Catalog": [],
+            "Number of Flares": [],
+            "Flare Latitude": [],
+            "Flare Longitude": [],
+            "Flare Start Time": [],
+            "Flare Peak Time": [],
+            "Flare End Time": [],
+            "Flare Last Data Time": [],
+            "Flare Intensity": [],
+            "Flare Integrated Intensity": [],
+            "Flare NOAA AR": [],
+            
+            "Predicted SEP All Clear": [],
+            "Predicted SEP Probability": [],
+            "Predicted SEP Threshold Crossing Time": [],
+            "Predicted SEP Start Time":[],
+            "Predicted SEP End Time": [],
+            "Predicted SEP Fluence": [],
+            "Predicted SEP Fluence Units": [],
+            "Predicted SEP Fluence Spectrum": [],
+            "Predicted SEP Fluence Spectrum Units": [],
+            "Predicted SEP Peak Intensity (Onset Peak)": [],
+            "Predicted SEP Peak Intensity (Onset Peak) Units": [],
+            "Predicted SEP Peak Intensity (Onset Peak) Time": [],
+            "Predicted SEP Peak Intensity Max (Max Flux)": [],
+            "Predicted SEP Peak Intensity Max (Max Flux) Units": [],
+            "Predicted SEP Peak Intensity Max (Max Flux) Time": [],
+            
+            "Predicted Point Intensity": [],
+            "Predicted Point Intensity Units": [],
+            "Predicted Point Intensity Time": [],
+
+            "Predicted Time Profile": []
+            
+            }
+
+    return dict
+
+
+def fill_forecast_dict_row(index, prediction, dict):
```

Comment:
This very long init function could be made shorter and more maintainable with a mapping dictionary and use of the built-in function `setattr()`.